### PR TITLE
Sync eden tests and add tpm and zfs workflows

### PR DESCRIPTION
**Pull request by giggsoff**
New Eden tests:

1. local app info
2. dev local ifo
3. publish location
4. shutdown test
5. app dhcp test
6. adam update to handle templates and test
7. zfs state and layout test
8. reduce local_datastore test time

Workflow updates:

1. Enable TPM in PR tests and remove separate building for ZFS
2. Enable ZFS in main branch tests
3. Update ubuntu and disable swap

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: .github/workflows/eden.yml [hunk 1]
```diff
@@ -10,11 +10,13 @@ on:
...
   integration:
-    runs-on: ubuntu-20.04
+    name: Integration test (tpm=${{ matrix.tpm }};${{ matrix.fs }})
+    runs-on: ubuntu-22.04
     strategy:
       fail-fast: false
       matrix:
-        hv: ["kvm", "zfs-kvm"]
+        tpm: ["true", "false"]
+        fs: ["zfs", "ext4"]
```
Comment (eriknordmark) [inline .github/workflows/eden.yml:19]: We're going from 2 to 4 tests for each PR and they currently take about 180 minutes. Will all 4 run in parallel?

Comment (giggsoff) [reply .github/workflows/eden.yml:19]: Yes, they are running in parallel runners.

**Comment by eriknordmark** [comment]
@giggsoff I don't know why the eden/integration test run was skipped; it was triggered by my approval, but then was skipped.

**Review by eriknordmark — APPROVED** [review]
Run eden again

**Comment by eriknordmark** [comment]
@giggsoff note that I changed one enum state name in https://github.com/lf-edge/eve/pull/2662 (from SHUTTING_DOWN to PREPARING_POWEROFF) and also added a PREPARED_POWEROFF. That requires a fix to the shutdown and devops tests but that doesn't seem to be why this is failing (unless I misread the log.)

**Review by eriknordmark — APPROVED** [review]
Run eden again

**Comment by giggsoff** [comment]
Well, one of the test failed with no enable file available, which I try to avoid with changes inside https://github.com/lf-edge/eve/pull/2636.

**Pull request merged**